crates/deltalake-core/src/delta_datafusion/mod.rs

```diff
@@ -586,8 +586,15 @@ impl<'a> DeltaScanBuilder<'a> {
         // However we may want to do some additional balancing in case we are far off from the above.
         let mut file_groups: HashMap<Vec<ScalarValue>, Vec<PartitionedFile>> = HashMap::new();
 
+        let table_partition_cols = self
```

Comment:
You can borrow here with `&` instead of cloning. Then you can pass that reference directory to `partitioned_file_from_action` instead of borrowing it for the call. It will save you the clone.

Reply:
great catch. I'll take the optimization.